Source/Core/DolphinQt/Debugger/RegisterWidget.cpp

```diff
@@ -269,6 +280,16 @@ void RegisterWidget::ShowContextMenu()
   menu->exec(QCursor::pos());
 }
 
+void RegisterWidget::AutoStep(const std::string& reg) const
+{
+  CodeTrace trace;
+  trace.SetRegTracked(reg);
+  trace.AutoStepping(true);
+  emit Host::GetInstance()->UpdateDisasmDialog();
+
+  // Could add timed out popup
```

Comment:
The return value of `AutoStepping` isn't checked and we don't have any way to know if it succeeded or not. Can't the method in `CodeViewWidget` be refactored and used here? Maybe in a member function of the `CodeTrace` class?

Reply:
I tacked the register autostep on as a quick way to navigate to a register in CodeView.  I can check for an error message, but I don't think it's necessary for succeeding.

Not sure what I would put in a CodeTrace member function to share the logic. I don't think I can do any Qt stuff in CodeTrace, since it's not a widget. Open to suggestions.